src/study/system-model/include/antares/study/system-model/component.h

```diff
@@ -72,6 +80,11 @@ class Component
         return data_.model;
     }
 
+    const Antares::Optimisation::LinearProblemApi::IScenario* getScenario() const
```

Comment:
actually, you need something like : 
```suggestion
    const Antares::Optimisation::LinearProblemApi::IScenario* getScenario(int MCyear) const
```
because the component should know which timeseries index to associate to every MC year